packages/apps/plugins/plugin-markdown/src/components/EditorMain.tsx

```diff
@@ -2,69 +2,94 @@
 // Copyright 2023 DXOS.org
 //
 
-import React, { type HTMLAttributes, useMemo, useEffect } from 'react';
+import { type EditorView } from '@codemirror/view';
+import React, { useMemo, useEffect } from 'react';
 
 import { LayoutAction, useIntentResolver } from '@dxos/app-framework';
-import { useTranslation } from '@dxos/react-ui';
+import { useRefCallback } from '@dxos/react-async';
+import { useThemeContext, useTranslation } from '@dxos/react-ui';
 import {
-  type TextEditorProps,
   type Comment,
-  MarkdownEditor,
+  type TextEditorProps,
+  TextEditor,
   Toolbar,
-  cursorLineMargin,
+  createBasicExtensions,
+  createMarkdownExtensions,
+  createThemeExtensions,
   editorFillLayoutRoot,
   editorFillLayoutEditor,
   focusComment,
   useComments,
-  useEditorView,
   useActionHandler,
   useFormattingState,
 } from '@dxos/react-ui-editor';
 import { attentionSurface, focusRing, mx, textBlockWidth } from '@dxos/react-ui-theme';
+import { nonNullable } from '@dxos/util';
 
 import { MARKDOWN_PLUGIN } from '../meta';
 
+// Expose editor view for playwright tests.
+// TODO(wittjosiah): Find a better way to expose this or find a way to limit it to test runs.
+const useTest = (view: EditorView | null) => {
+  useEffect(() => {
+    const composer = (window as any).composer;
+    if (composer) {
+      composer.editorView = view;
+    }
+  }, [view]);
+};
+
 export type EditorMainProps = {
-  comments?: Comment[];
+  id: string;
+  readonly?: boolean;
   toolbar?: boolean;
-} & Pick<TextEditorProps, 'model' | 'readonly' | 'extensions'>;
+  comments?: Comment[];
+} & Pick<TextEditorProps, 'doc' | 'extensions'>;
 
-export const EditorMain = ({ model, comments, toolbar, extensions: _extensions, ...props }: EditorMainProps) => {
+export const EditorMain = ({ id, readonly, toolbar, comments, doc, extensions: _extensions }: EditorMainProps) => {
   const { t } = useTranslation(MARKDOWN_PLUGIN);
+  const { themeMode } = useThemeContext();
 
-  const [editorRef, viewInvalidated] = useEditorView(model.id);
-  useComments(viewInvalidated ? null : editorRef.current, model.id, comments);
-  const handleAction = useActionHandler(editorRef.current);
+  const { ref: editorRef, value: view } = useRefCallback<EditorView>();
```

Comment:
If this is the best or only way to get `view`, I think it’d be easier to proliferate this pattern if `react-ui-editor` exported its own `useEditorRef` hook.

Reply:
i debated that but given that it's only used once and i want to try to remove the component based approach altogether decided against. for now.

Reply:
Makes sense 👌